pydatalab/tests/server/test_samples.py

```diff
@@ -750,3 +750,186 @@ def test_add_items_to_collection_success(client, default_collection, example_ite
+    assert response.status_code == 201
+
+    collection_from_db = database.collections.find_one({"collection_id": "test_collection_remove"})
+    collection_object_id = collection_from_db["_id"]
+
+    item_ids = [sample_1_dict["item_id"], sample_2_dict["item_id"]]
+
+    for item_id in item_ids:
+        database.items.update_one(
+            {"item_id": item_id},
+            {
+                "$push": {
+                    "relationships": {
+                        "type": "collections",
+                        "immutable_id": collection_object_id,
+                    }
+                }
+            },
+        )
+
+    for item_id in item_ids:
+        item = database.items.find_one({"item_id": item_id})
+        assert item is not None
+        collection_relationships = [
+            rel for rel in item.get("relationships", []) if rel.get("type") == "collections"
+        ]
+        assert len(collection_relationships) == 1
+
+    item_1_refcode = client.get(f"/get-item-data/{sample_1_dict['item_id']}").json["item_data"][
+        "refcode"
+    ]
+    item_2_refcode = client.get(f"/get-item-data/{sample_2_dict['item_id']}").json["item_data"][
+        "refcode"
+    ]
+    refcodes = [item_1_refcode, item_2_refcode]
+
+    response = client.delete(
+        "/collections/test_collection_remove/items", json={"refcodes": refcodes}
+    )
+
+    assert response.status_code == 200
+    data = response.get_json()
+    assert data["status"] == "success"
+    assert data["removed_count"] == 2
+
+    for item_id in item_ids:
+        item = database.items.find_one({"item_id": item_id})
+        assert item is not None
+        collection_relationships = [
+            rel for rel in item.get("relationships", []) if rel.get("type") == "collections"
+        ]
+        assert len(collection_relationships) == 0
+
+
+@pytest.mark.dependency()
+def test_remove_items_from_collection_not_found(client):
+    """Test removing items from non-existent collection."""
+    response = client.delete(
+        "/collections/nonexistent_collection/items", json={"refcodes": ["refcode1", "refcode2"]}
+    )
+
+    assert response.status_code == 404
+    data = response.get_json()
+    assert data["error"] == "Collection not found"
+
+
+@pytest.mark.dependency()
+def test_remove_items_from_collection_no_items_provided(client, default_collection):
+    """Test removing with no item IDs provided."""
+    collection_dict = default_collection.dict()
+    collection_dict["collection_id"] = "test_collection_empty_items"
+    response = client.put("/collections", json={"data": collection_dict})
+    assert response.status_code == 201
+
+    collection_id = collection_dict["collection_id"]
+    response = client.delete(f"/collections/{collection_id}/items", json={"refcodes": []})
+
+    assert response.status_code == 400
+    data = response.get_json()
+    assert data["error"] == "No refcodes provided"
```

Comment:
Just a comment that this made me realise the collections routes do not use standard errors (i.e., status = "error", then "message" = "whatever"). Not something to fix in this PR, but we should go through and unify all routes on this at some point.